src/is_wire/core/message.py

```diff
@@ -0,0 +1,363 @@
+        was serialized """
+        return self._content_type
+
+    @content_type.setter
+    def content_type(self, value):
+        assert_type(value, ContentType, "content_type")
+        self._content_type = value
+
+    def has_content_type(self):
+        """ Returns: True if the property content_type of the message is set,
+         False otherwise """
+        return self._content_type is not None
+
+    # created_at
+
+    @property
+    def created_at(self):
+        """ float: Seconds since the epoch indicating when the message
+        was created"""
+        return self._created_at
+
+    @created_at.setter
+    def created_at(self, timestamp):
+        # assert_type(timestamp, (), "created_at")
+        self._created_at = timestamp
+
+    def has_created_at(self):
+        """ Returns: True if the property created_at of the message is set,
+         False otherwise """
+        return self._created_at is not None
+
+    # metadata
+
+    @property
+    def metadata(self):
+        """ dict: Key-value pairs which any can model any type of extra
+        information about the message """
+        return self._metadata
+
+    @metadata.setter
+    def metadata(self, value):
+        assert_type(value, dict, "metadata")
+        self._metadata = value
+
+    def has_metadata(self):
+        """ Returns: True if the property metadata of the message is set,
+         False otherwise """
+        return len(self._metadata) != 0
+
+    # timeout
+
+    @property
+    def timeout(self):
+        return self._timeout
+
+    @timeout.setter
+    def timeout(self, seconds):
+        assert_type(seconds, [float] + list(integer_types), "timeout")
+        self._timeout = seconds
+
+    def has_timeout(self):
+        """ Returns: True if the property timeout of the message is set,
+         False otherwise """
+        return self._timeout is not None
+
+    def deadline_exceeded(self):
+        if not self.has_timeout():
+            return False
+        return now() > self.created_at + self.timeout
+
+    # status
+
+    @property
+    def status(self):
+        """ Status representing the success or not of a RPC """
+        return self._status
+
+    @status.setter
+    def status(self, value):
+        assert_type(value, Status, "status")
+        self._status = value
+
+    def has_status(self):
+        """ Returns: True if the property status of the message is set,
+         False otherwise """
+        return self._status is not None
+
+    # tracing
+
+    def extract_tracing(self):
+        return TextFormatPropagator.from_carrier(self.metadata)
+
+    def inject_tracing(self, span):
+        span_context = TextFormatPropagator.new_span_context(
+            trace_id=span.context_tracer.trace_id,
+            span_id=span.span_id,
+        )
+        self.metadata = TextFormatPropagator.to_carrier(
+            span_context, self.metadata)
+
+    # pack / unpack
+
+    def pack(self, obj):
```

Comment:
Due the example below where `content_type` is set after `pack` call,
```python
from is_wire.core import Message, ContentType
from google.protobuf.struct_pb2 import Struct

struct = Struct()
struct.fields["apples"].string_value = "red"

m = Message()
m.topic = 'MyStruct'
m.pack(struct)
m.content_type = ContentType.JSON
struct2 = m.unpack(Struct) # got an exception here!
```
I suggest to remove the setter method of `content_type`, turning this an option to the pack method with `ContentType.PROTOBUF` as default.

Reply:
In case of accept this change, remember to change the 3rd readme example.

Reply:
I think the use case is not enough to justify removing the setter. If we intend to create a more generic library that can accomodated the use case of others I think this field should stay for now. For instance if the user wants to serialize something as json but using the standard library instead of a protobuf schema (through pack) they will need to set the content_type field.